pom.xml

```diff
@@ -76,5 +76,10 @@ THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
             <version>1.10.19</version>
             <scope>test</scope>
         </dependency>
+        <dependency>
+            <groupId>com.google.guava</groupId>
```

Comment:
Do we need to make this library have this explicit dependency just for an ImmutableMap / Set? Does there have to be a strong contract on the Map / Set that it is Immutable?